### ISSUE TITLE
Turn on gwt facet on new app

### ISSUE THREAD

**Issue by branflake2267** [gwt-plugins/gwt-eclipse-plugin#224]
Should I install the facet, but this means facets need to be turned on. In that case, I think leave it up to dev.

**Comment by johnthad** [comment]
When I try to turn on the GWT facet, Eclipse Neon says that it cannot find the SDK, whether I point to the 2.7.0 plugin or the 2.7.0 installed in my /opt directory. In the facets directory I also get an error message that GWT requires Java 1.5.0 or better even though I select the Java 1.6 runtime. (Frankly, facets are one of those parts of Eclipse that I dunno has _ever_ worked in the 10 years I've used GWT and Eclipse. They've always caused me trouble--on Mac, Windows, or Linux.)

**Comment by branflake2267** [comment]
I can understand the challenge with facets, but this is one of the things that has been improved in Eclipse. I suspect it's a facet configuration issue in the GWT plugin. Facets don't have to be used, although it makes it much easier to do automated things with some of the tooling that is coming. Such as the Google Cloud tools that is coming. I'm hoping to get it dialed in so easier debugging can be had with some of the server runtime features.

* What version of Eclipse are you using? Mars, Neon? And is it Java EE? 
* Are you using maven or is it a standard Java project?

**Comment by johnthad** [comment]
I am running "Eclipse IDE for Java Developers / Version: Neon.1a Release (4.6.1) / Build id: 20161007-1200" on two Macs, one Yosemite, the other El Capitan. I'm using Java 1.6, Gradle 2.14.1, and Steffan Schaefer's Gradle plugin (https://github.com/steffenschaefer/gwt-gradle-plugin).

**Comment by branflake2267** [comment]
Thanks, I haven't tested with that plugin yet. I'll test beginning of next week.

**Comment by branflake2267** [comment]
What's the best way to setup a test case, do you have a project template I could use to setup project?

**Comment by johnthad** [comment]
I don't have a project template. I'll see if I can recreate the error with the "getting started" project or one of the tutorials. (I only have 3 GWT-based products. The first to use Gradle has never had facets turned on. I only converted this app from Ant to Gradle last month. The third app uses Ant.)

There's been no change to Steffan's plugin in quite a while, and some speculate that he's stopped development. I briefly tried the Putnami plugin (https://github.com/Putnami/putnami-gradle-plugin) and liked it. However it requires Java 7 or better (it uses Jetty 9), and I found that Java 7 JARs were slipping into the WEB-INF/lib and going unnoticed. I'm stuck with Java 6 because of a major customer. Eventually I purged those JARs, but it's safer if my dev environment reflects the minimum deployed environment.

**Comment by branflake2267** [comment]
Gotcha. There isn't a need for the GWT facet, it doesn't have to be turned on. I'm planning on adding Gradle support sometime after GWT Con. I'm trying to stabilize everything before adding more.

**Comment by branflake2267** [comment]
Just an fyi... I started testing, I haven't seen the issue yet. I tried a couple basic project configurations, Java and Java Ant. Did you happen to upgrade or did you use a clean install? Can I get a screenshot of your facets? 

![screen shot 2016-10-14 at 8 57 42 pm](https://cloud.githubusercontent.com/assets/1326504/19407247/e28da650-9250-11e6-9df0-bef5b45f01ed.png)

**Comment by branflake2267** [comment]
I've removed the project configurator logic, so things should clear up on the next push. Its simply just a marker that GWT is being used now. I'll move the other logic to project configurators. New projects from will turn the facet on if wtp is installed.

**Comment by branflake2267** [comment]
Just to confirm it's not required, except with server runtimes.

**Comment by johnthad** [comment]
If I turn on facets for GWT and Java (see attached) I get over 4,700 errors in 60K lines of code. Turn them off and there are no warnings or errors.

![screen shot 2016-10-15 at 10 52 50](https://cloud.githubusercontent.com/assets/5176962/19410959/1727b9b2-92c6-11e6-951e-683c4a2408ef.png)

**Comment by branflake2267** [comment]
Thanks for the info. I've removed the configurator pieces so the errors should go away. I should have things worked out by end of Sunday.

**Comment by branflake2267** [comment]
https://github.com/gwt-plugins/gwt-eclipse-plugin/pull/290

**Comment by johnthad** [comment]
See https://github.com/johnthad/MyWebApp
(Sorry this took so long.)

**Comment by branflake2267** [comment]
Cool thanks. I see some manual controls in the pom.xml, I suspect that's why there's a pom.xml to control m2e. Why is there a gradle jar?

**Comment by branflake2267** [comment]
I'm going to move this to a new issue.

**Comment by branflake2267** [comment]
https://github.com/gwt-plugins/gwt-eclipse-plugin/issues/293

**Comment by johnthad** [comment]
The Gradle JAR file part of the [Gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html). I _much prefer_ Gradle to Maven, but the Gradle folks update the thing at a dizzying rate. Using the wrapper ensures that your project keeps (and shares) the same Gradle version across users. 

The first time to run `./gradlew tasks` (or any Gradle task), the Gradle version that matches your wrapper will download to `~/.gradle/wrapper/dists`. Others building your project do not have to download Gradle or fret the version--it's done for them.

**Comment by branflake2267** [comment]
Thanks for that info. It should be pretty easy to add the Gradle project configurator, but I'll plan on that after GWTCon.

**Comment by johnthad** [comment]
With plugin version 3.0.0.201610170319, I am able to turn on the GWT and Java 6 facets with no errors. I still dunno I _do_ with them, but there are no errors. :)  Thanks.

**Comment by branflake2267** [comment]
Good to hear. The facet is used with the WTP server runtimes. If you start a server runtime, I can do things like start GWT Development Mode without user intervention. The goal is to be able to provide more opportunities with external servers, different server runtimes (web servers), and probably mobile. I don't have it all done yet, I'm waiting on another patch to see how things move forward

**Comment by branflake2267** [comment]
Closing this issue. Create a new issue if there are any new issues.
